jinja2/filters.py

```diff
@@ -977,10 +984,17 @@ def do_map(*args, **kwargs):
 
     .. versionadded:: 2.7
     """
+    try:
```

Comment:
This should probably be done in `prepare_map`.

Reply:
See commit below, moved to prepare_map